modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala

```diff
@@ -353,7 +353,7 @@ package object engine {
 
   def pure[A](a: A): HandleP[A] = Applicative[HandleP].pure(a)
 
-  private val unit: HandleP[Unit] = pure(Unit)
+  private val unit: HandleP[Unit] = pure(())
```

Comment:
Above you were using the companion object of `trait Unit`, called `Unit`, which is not of type `Unit` … it's type is the unrelated `Unit.type`. In case you weren't already confused.